packages/client/src/runtime/core/types/GetResult.ts

```diff
@@ -50,19 +50,18 @@ type Count<O> = { [K in keyof O]: Count<number> } & {}
 // prettier-ignore
 export type GetFindResult<P extends Payload, A> =
   {} extends A ? DefaultSelection<P> :
-  A extends 
-  | { select: infer S } & Record<string, unknown>
-  | { include: infer S } & Record<string, unknown>
-  ? S extends undefined ? DefaultSelection<P> :
-    {
-      [K in keyof S as S[K] extends false | undefined | null ? never : K]:
-        S[K] extends object
+  A extends
+  | { select: infer S extends object } & Record<string, unknown>
+  | { include: infer I extends object } & Record<string, unknown>
+  ? {
+      [K in keyof S | keyof I as (S & I)[K] extends false | undefined | null ? never : K]:
+        (S & I)[K] extends object
         ? P extends SelectablePayloadFields<K, (infer O)[]>
-          ? O extends Payload ? GetFindResult<O, S[K]>[] : never
+          ? O extends Payload ? GetFindResult<O, (S & I)[K]>[] : never
```

Comment:
I am curious if that will now produce unexpected result:

```ts
include: {
   field: true
}
select: {
   field: false
}
```

Since we can't intersect `true & false`?

Reply:
This case is in theory protected against on user input when both are provided and not optional. However, that does not apply to utility types. In that case, as you pointed out, the intersection is non-valid and the whole mapped type has no keys to map on, and resolves to `{}`. Then because `include` is present, the default selection is added to it.

Reply:
Not sure if this is something to test or document. Perhaps we should test it, as it is currently a bit undefined in behavior. Or perhaps it does not matter because it is invalid? wdyt

Reply:
I think we sort of have a test for `select` + `include` being invalid [here](https://github.com/prisma/prisma/blob/19fd17d21c9e2e49e5b4640deb8b355dbbe1246f/packages/client/tests/functional/query-validation/tests.ts#L10) and `@ts-expect-error` ensures we have a compile-time error as well, so I think we'd be fine without `GetResult`-specific test.